src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -2,18 +2,43 @@
 
 import core.basesyntax.Storage;
 
+@SuppressWarnings("unchecked")
+
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int NOT_FOUND = -1;
+    private static final int MAX_SIZE = 10;
+    private int size;
+    private Pair<K, V>[] items = new Pair[MAX_SIZE];
+
     @Override
     public void put(K key, V value) {
+        int index;
+        if ((index = findIndexByKey(key)) != NOT_FOUND) {
+            items[index].setValue(value);
+            return;
+        }
+        items[size] = new Pair<>(key, value);
+        size++;
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int index;
+        return (index = findIndexByKey(key)) == NOT_FOUND ? null : items[index].getValue();
```

Comment:
```suggestion
        return index = findIndexByKey(key) == NOT_FOUND ? null : items[index].getValue();
```
would look better if you initialize `index` variable on line 26